qa/rpc-tests/test_framework/mininode.py

```diff
@@ -994,14 +995,16 @@ def deserialize(self, f):
         self.message = deser_string(f)
         self.code = struct.unpack("<B", f.read(1))[0]
         self.reason = deser_string(f)
-        if (self.message == "block" or self.message == "tx"):
+        if (self.code != self.REJECT_MALFORMED and
+                (self.message == "block" or self.message == "tx")):
             self.data = deser_uint256(f)
 
     def serialize(self):
         r = ser_string(self.message)
         r += struct.pack("<B", self.code)
         r += ser_string(self.reason)
-        if (self.message == "block" or self.message == "tx"):
+        if (self.code != self.REJECT_MALFORMED and
+                (self.message == "block" or self.message == "tx")):
```

Comment:
I have all of this included in my py3 branch, so it is not mandatory to fix all issues right now. (Otherwise I'd be creating pulls every other day to keep the whole py2 codebase "updated") E.g. we got new test in `fundrawtransaction.py` which are using the old `except JSONRPCException,e:` syntax but it is not worth to create a pull for those right now, imo.